### ISSUE TITLE
wxPropertyGrid is not rendered correctly when scrolling under wxQt

### ISSUE THREAD

**Issue by AliKet** [wxWidgets/wxWidgets#24136]
#### Bug description:

The scrolled-in area remains black as shown in the image below:
![Screenshot_20231214_143108](https://github.com/wxWidgets/wxWidgets/assets/7704771/9cfa171e-b6cc-47c0-9913-164d21788e47)

#### To Reproduce:

1. Run 'propgrid' sample
2. Scroll down
3. See the bug

Unfortunately the issue is still not resolved even with PR #24135

My little knowledge of `wxPropertyGrid`'s internals makes it more difficult for me to try to resolve this issue. Sorry!

### Platform and version information

- wxWidgets version  you use: lateset and v3.1.7 (commit 7ad1bffa87)
- wxWidgets port  you use: wxQt (Qt 5.9.5)
- OS  and its version: Kubuntu 18.04

**Comment by AliKet** [comment]
Fortunatly the fix is simple: we should override `IsDoubleBuffered()`  to return true because under wxQt it is always true. I'll include the fix in #24135

**Comment by AliKet** [comment]
@vadz @a-wi How about using `std::unique_ptr` here instead of raw pointer:
https://github.com/wxWidgets/wxWidgets/blob/6d89a14c7256e9d4723f3b212a760ce0152fe7e5/src/propgrid/propgrid.cpp#L1880-L1928

Something like this:
```
diff --git a/src/propgrid/propgrid.cpp b/src/propgrid/propgrid.cpp
index 0d38410cb7..b6b05c351e 100644
--- a/src/propgrid/propgrid.cpp
+++ b/src/propgrid/propgrid.cpp
@@ -1879,20 +1879,20 @@ wxPGProperty* wxPropertyGrid::DoGetItemAtY( int y ) const
 
 void wxPropertyGrid::OnPaint( wxPaintEvent& WXUNUSED(event) )
 {
-    wxDC* dcPtr = nullptr;
+    std::unique_ptr<wxDC> dcPtr;
     if ( !HasExtraStyle(wxPG_EX_NATIVE_DOUBLE_BUFFERING) )
     {
         if ( m_doubleBuffer )
         {
-            dcPtr = new wxBufferedPaintDC(this, *m_doubleBuffer);
+            dcPtr.reset( new wxBufferedPaintDC(this, *m_doubleBuffer) );
         }
     }
     if ( !dcPtr )
     {
-        dcPtr = new wxPaintDC(this);
+        dcPtr.reset( new wxPaintDC(this) );
     }
     wxASSERT( dcPtr );
-    PrepareDC(*dcPtr);
+    PrepareDC(*dcPtr.get());
     // Unused area will be cleared when drawing the items
 /*
     dcPtr->SetBackground(m_colEmptySpace);
@@ -1902,7 +1902,6 @@ void wxPropertyGrid::OnPaint( wxPaintEvent& WXUNUSED(event) )
     // Don't paint after destruction has begun
     if ( !HasInternalFlag(wxPG_FL_INITIALIZED) )
     {
-        delete dcPtr;
         return;
     }
 
@@ -1919,9 +1918,6 @@ void wxPropertyGrid::OnPaint( wxPaintEvent& WXUNUSED(event) )
     // Repaint this rectangle
     DrawItems(*dcPtr, r.y, r.y + r.height-1, &r);
 
-    // This blits the buffer (if used) to the window DC.
-    delete dcPtr;
-
     // We assume that the size set when grid is shown
     // is what is desired.
     SetInternalFlag(wxPG_FL_GOOD_SIZE_SET);

```

**Comment by a-wi** [comment]
And the issue is gone after using std::unique_ptr?

**Comment by vadz** [comment]
I don't think the proposed change actually changes anything, but it makes the code shorter and safer, so I'd still be for applying it.

The actual issue seems to be fixed by overriding `IsDoubleBuffered()`, as mentioned above.

**Comment by AliKet** [comment]
> And the issue is gone after using std::unique_ptr?

Sorry, the issue wasn't `wxPropertyGrid`'s fault at all in the first place. and is now corrected by this commit [cf52a6c](https://github.com/AliKet/wxWidgets/commit/cf52a6cee532589700ebc5db7e58040132b376f3)
because `wxQt` is double buffered by default since v4.0. but the library was tricked into using `wxBufferedPaintDC` instead of `wxPaintDC` in the paint event handler because it was not told to do so.

As for the std::unique_ptr suggestion, it is as vadz commented above. sorry for the confusion!

**Comment by a-wi** [comment]
I think using std::unique_ptr is fine here.
